modules/test/playwright/tests/object-web/objectFields.spec.ts

```diff
@@ -56,7 +56,68 @@ test.describe('Manage object fields through Model Builder', () => {
 		);
 
 		await apiHelpers.listTypeAdmin.deleteListTypeDefinition(
-			ListTypeDefinition.id
+			listTypeDefinition.id
 		);
 	});
+
+	test('all picklist definitions are listed during object field creation', async ({
+		apiHelpers,
+		modelBuilderPage,
+		page,
+		viewObjectDefinitionsPage,
+	}) => {
+		const listTypeDefinitions = await Promise.all(
+			Array(22)
+				.fill(null)
+				.map(() =>
+					apiHelpers.listTypeAdmin.postRandomListTypeDefinition()
+				)
+		);
+
+		const objectDefinition =
+			await apiHelpers.objectAdmin.postRandomObjectDefinition('default');
+
+		try {
+			await page.goto('/');
+
+			await viewObjectDefinitionsPage.goto();
+
+			await viewObjectDefinitionsPage.openObjectFolder('default');
+
+			await viewObjectDefinitionsPage.viewInModelBuilder();
+
+			await modelBuilderPage.openObjectFieldSelectionPage(
+				objectDefinition.name,
+				'Picklist',
+				'objectFieldLabel' + getRandomInt()
+			);
+
+			modelBuilderPage.newObjectFieldSelectPicklist.click();
+
+			const listTypeDefinitionBox =
+				modelBuilderPage.page.getByRole('listbox');
+
+			await expect(listTypeDefinitionBox).toBeVisible();
+
+			await expect(
+				listTypeDefinitionBox.getByRole('listitem')
+			).toHaveCount(22);
+		}
+		finally {
```

Comment:
To ensure cleanup, please use a `afterEach` hook,  something like this:

```
const createdEntities = {listTypeDefinitionIds: []};

test.afterEach(async ({apiHelpers, scriptManagementPage}) => {
	if (createdEntities.listTypeDefinitionIds.length) {
		await Promise.all(
			listTypeDefinitions.map((listTypeDefinition) =>
				apiHelpers.listTypeAdmin.deleteListTypeDefinition(
					listTypeDefinition.id
				)
			)
		);

		createdEntities.listTypeDefinitionIds = [];
	}
});
```

For this to work you need to `createdEntities.listTypeDefinitionIds.push(objectDefinition.id)` after every entity created.

Reply:
We tend to prefer to use a Playwright-native way of doing things, nothing wrong with try / catch / finally.